src/baggage-span-processor.ts

```diff
@@ -0,0 +1,43 @@
+import { Context, propagation, Span } from '@opentelemetry/api';
+import { SpanProcessor } from '@opentelemetry/sdk-trace-base';
+
+/**
+ * A {@link SpanProcessor} that reads entries stored in {@link Baggage}
+ * from the parent context and adds the baggage entries' keys and values
+ * to the span as attributes on span start.
+ *
+ * Keys and values added to Baggage will appear on subsequent child
+ * spans for a trace within this service *and* be propagated to external
+ * services in accordance with any configured propagation formats
+ * configured. If the external services also have a Baggage span
+ * processor, the keys and values will appear in those child spans as
+ * well.
+ *
+ * ⚠ Warning ⚠️
```

Comment:
This whole warning came from the wording we had in the node distro. Is it too intense or is it just right? 🤷

Reply:
I'd say you could go even _more_ intense if the mood suited you. Emoji explosion. 

and/or maybe we could add some spicy logs when debug mode is active?

```🚨🚨🚨🚨🚨🚨 ADDED xxxx TO BAGGAGE 🚨🚨🚨🚨🚨🚨```
```BROADCASTING PLAIN VALUE IN ALL HEADERS```

i love that you're calling it out

Reply:
Now if debug is enabled this shows up in the console:
![baggage in all outgoing headers](https://github.com/honeycombio/honeycomb-opentelemetry-web/assets/29520003/f69f9d9e-bf53-4fc8-8806-a6223c03a8c0)

Reply:
😻 😻 😻